src/com/amazon/ion/impl/IonReaderContinuableApplicationBinary.java

```diff
@@ -239,6 +514,43 @@ String getSymbol(int sid) {
         return symbols[localSid];
     }
 
+    /**
+     * Creates a SymbolToken representation of the given symbol ID.
+     * @param sid a symbol ID.
+     * @return a SymbolToken.
+     */
+    private SymbolToken getSymbolToken(int sid) {
+        int symbolTableSize = localSymbolMaxId + firstLocalSymbolId + 1; // +1 because the max ID is 0-indexed.
+        if (symbolTokensById == null) {
+            symbolTokensById = new ArrayList<>(symbolTableSize);
+        }
+        if (symbolTokensById.size() < symbolTableSize) {
+            for (int i = symbolTokensById.size(); i < symbolTableSize; i++) {
+                symbolTokensById.add(null);
+            }
+        }
```

Comment:
Assuming we no longer have to worry about nullability for `symbolTokensById`:

```suggestion
        // Avoid redundant list growth operations, if it comes to that
        symbolTokensById.ensureCapacity(symbolTableSize);
        if (symbolTokensById.size() < symbolTableSize) {
            for (int i = symbolTokensById.size(); i < symbolTableSize; i++) {
                // We rely on eager null initialization of the backing list to provide map semantics
                symbolTokensById.add(null);
            }
        }
```
We can avoid redundant growth events, if it comes to that.